matrix/Quaternion.hpp

```diff
@@ -330,24 +330,38 @@ class Quaternion : public Vector<Type, 4>
 
     /**
      * Rotate quaternion from rotation vector
-     * TODO replace with AxisAngle call
      *
      * @param vec rotation vector
      */
-    void rotate(const Vector<Type, 3> &vec)
+    void rotate(const AxisAngle<Type> &vec)
     {
-        Quaternion res;
-        res.from_axis_angle(vec);
-        (*this) = (*this) * res;
+        Quaternion res(vec);
+        (*this) = res * (*this);
     }
 
+    /**
+     * Rotates vector v_1 in frame 1 to vector r_2 in frame 2
+     * using the rotation quaternion q_12
+     * v_2 = q_12 * v_1 * q_12^-1
+     *
```

Comment:
what is exactly the convention for the naming: q_12 vs q_21

Reply:
You are totally right, the indexing convention is widespread the opposite way (also in the paper I'm referncing in the original post). I'll change the comments.